# 큰 DataFrame 살펴보기

In [1]:
import pandas as pd

In [7]:
laptops_df = pd.read_csv('laptops.csv')
laptops_df

brand             model  ram hd_type  hd_size  screen_size   price  \
0     Dell  Inspiron 15-3567    4     hdd     1024         15.6   40000   
1    Apple       MacBook Air    8     ssd      128         13.3   55499   
2    Apple       MacBook Air    8     ssd      256         13.3   71500   
3    Apple       MacBook Pro    8     ssd      128         13.3   96890   
4    Apple       MacBook Pro    8     ssd      256         13.3  112666   
..     ...               ...  ...     ...      ...          ...     ...   
162   Asus            A555LF    8     hdd     1024         15.6   39961   
163   Asus     X555LA-XX172D    4     hdd      500         15.6   28489   
164   Asus            X554LD    2     hdd      500         15.6   29199   
165   Asus    X550LAV-XX771D    2     hdd      500         15.6   29990   
166   Asus     X540LA-XX538T    4     hdd     1024         15.6   30899   

    processor_brand processor_model  clock_speed graphic_card_brand  \
0             intel              i5          2.5              intel   
1             intel              i5          1.8              intel   
2             intel              i5          1.8              intel   
3             intel              i5          2.3              intel   
4             intel              i5          2.3              intel   
..              ...             ...          ...                ...   
162           intel      i3 4th gen          1.7             nvidia   
163           intel      i3 4th gen          1.9              intel   
164           intel      i3 4th gen          1.9              intel   
165           intel      i3 4th gen          1.7              intel   
166           intel      i3 5th gen          2.0              intel   

     graphic_card_size       os  weight comments  
0                  NaN    linux    2.50      NaN  
1                  2.0      mac    1.35      NaN  
2                  2.0      mac    1.35      NaN  
3                  2.0      mac    3.02      NaN  
4                  2.0      mac    3.02      NaN  
..                 ...      ...     ...      ...  
162                2.0  windows    2.30      NaN  
163                NaN    linux    2.30      NaN  
164                1.0    linux    2.30      NaN  
165                NaN    linux    2.50      NaN  
166                NaN  windows    2.30      NaN  

[167 rows x 15 columns]

### 데이터 위에 몇개만 보고 싶다면
* `df.head(int)` 를 사용한다.

In [4]:
laptops_df.head(3)

brand             model  ram hd_type  hd_size  screen_size  price  \
0   Dell  Inspiron 15-3567    4     hdd     1024         15.6  40000   
1  Apple       MacBook Air    8     ssd      128         13.3  55499   
2  Apple       MacBook Air    8     ssd      256         13.3  71500   

  processor_brand processor_model  clock_speed graphic_card_brand  \
0           intel              i5          2.5              intel   
1           intel              i5          1.8              intel   
2           intel              i5          1.8              intel   

   graphic_card_size     os  weight comments  
0                NaN  linux    2.50      NaN  
1                2.0    mac    1.35      NaN  
2                2.0    mac    1.35      NaN

In [5]:
laptops_df.head(10)

brand                   model  ram hd_type  hd_size  screen_size   price  \
0   Dell        Inspiron 15-3567    4     hdd     1024         15.6   40000   
1  Apple             MacBook Air    8     ssd      128         13.3   55499   
2  Apple             MacBook Air    8     ssd      256         13.3   71500   
3  Apple             MacBook Pro    8     ssd      128         13.3   96890   
4  Apple             MacBook Pro    8     ssd      256         13.3  112666   
5  Apple  MacBook Pro (TouchBar)   16     ssd      512         15.0  226000   
6  Apple  MacBook Pro (TouchBar)   16     ssd      512         13.3  158000   
7  Apple                MacBook     8     ssd      256         12.0   96990   
8   Dell             Vostro 3468    4     hdd     1024         14.0   33225   
9   Dell      Inspiron 15.6 3552    4     hdd      500         15.0   21990   

  processor_brand processor_model  clock_speed graphic_card_brand  \
0           intel              i5          2.5              intel   
1           intel              i5          1.8              intel   
2           intel              i5          1.8              intel   
3           intel              i5          2.3              intel   
4           intel              i5          2.3              intel   
5           intel              i7          2.7              intel   
6           intel              i5          2.9              intel   
7           intel          core m          1.2              intel   
8           intel              i3          2.4              intel   
9           intel   pentium N3710          1.6              intel   

   graphic_card_size     os  weight             comments  
0                NaN  linux    2.50                  NaN  
1                2.0    mac    1.35                  NaN  
2                2.0    mac    1.35                  NaN  
3                2.0    mac    3.02                  NaN  
4                2.0    mac    3.02                  NaN  
5                2.0    mac    2.50                  NaN  
6                2.0    mac    1.37                  NaN  
7                2.0    mac    0.92                  NaN  
8                NaN  linux    1.76  Integrated Graphics  
9                NaN  linux    2.19  Integrated Graphics

### 데이터 아래 몇개만 보고 싶다면
* `df.tail(int)` 를 사용한다.

In [6]:
laptops_df.tail(3)

brand           model  ram hd_type  hd_size  screen_size  price  \
164  Asus          X554LD    2     hdd      500         15.6  29199   
165  Asus  X550LAV-XX771D    2     hdd      500         15.6  29990   
166  Asus   X540LA-XX538T    4     hdd     1024         15.6  30899   

    processor_brand processor_model  clock_speed graphic_card_brand  \
164           intel      i3 4th gen          1.9              intel   
165           intel      i3 4th gen          1.7              intel   
166           intel      i3 5th gen          2.0              intel   

     graphic_card_size       os  weight comments  
164                1.0    linux     2.3      NaN  
165                NaN    linux     2.5      NaN  
166                NaN  windows     2.3      NaN

### 데이터를 알고 싶다면
* 크기를 알고 싶다면 `df.shape` 를 찍어본다

In [8]:
laptops_df.shape

(167, 15)

* column들을 알고 싶다면 `df.colums` 를 찍어본다

In [10]:
laptops_df.columns

Index(['brand', 'model', 'ram', 'hd_type', 'hd_size', 'screen_size', 'price',
       'processor_brand', 'processor_model', 'clock_speed',
       'graphic_card_brand', 'graphic_card_size', 'os', 'weight', 'comments'],
      dtype='object')

* column들의 기본 정보를 알고 싶다면 `df.info()` 를 찍어보기
* row 수는 167 개인데, 167 개가 안되는 것들은 `비어있는 값`이라고 볼 수 있다.

In [13]:
laptops_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 167 entries, 0 to 166
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   brand               167 non-null    object 
 1   model               167 non-null    object 
 2   ram                 167 non-null    int64  
 3   hd_type             167 non-null    object 
 4   hd_size             167 non-null    int64  
 5   screen_size         167 non-null    float64
 6   price               167 non-null    int64  
 7   processor_brand     167 non-null    object 
 8   processor_model     167 non-null    object 
 9   clock_speed         166 non-null    float64
 10  graphic_card_brand  163 non-null    object 
 11  graphic_card_size   81 non-null     float64
 12  os                  167 non-null    object 
 13  weight              160 non-null    float64
 14  comments            55 non-null     object 
dtypes: float64(4), int64(3), object(8)
memory usage: 19.7+ KB

* columns의 통계 값을 알고 싶어!
* `df.describe()`

In [15]:
laptops_df.describe()

ram     hd_size  screen_size          price  clock_speed  \
count  167.000000   167.00000   167.000000     167.000000   166.000000   
mean     6.898204   768.91018    14.775210   64132.898204     2.321084   
std      3.787479   392.99080     1.376526   42797.674010     0.554187   
min      2.000000    32.00000    10.100000   13872.000000     1.100000   
25%      4.000000   500.00000    14.000000   35457.500000     1.900000   
50%      8.000000  1024.00000    15.600000   47990.000000     2.300000   
75%      8.000000  1024.00000    15.600000   77494.500000     2.600000   
max     16.000000  2048.00000    17.600000  226000.000000     3.800000   

       graphic_card_size      weight  
count          81.000000  160.000000  
mean           52.160494    2.250813  
std           444.134142    0.648446  
min             1.000000    0.780000  
25%             2.000000    1.900000  
50%             2.000000    2.200000  
75%             4.000000    2.600000  
max          4000.000000    4.200000

* 기준을 정해서 정렬하고 싶다면
* `df.sort_values(by='정렬기준')` (기본값 오름차순/ 내림차순으로 하고 싶다면 `ascending=False`)
* 원본이 변하는 것이 아니라 새로운 df가 생기는 것
* 기존 원본을 바꾸고 싶다면 `inplace=True`

In [18]:
laptops_df.sort_values(by="price")

brand                       model  ram hd_type  hd_size  screen_size  \
148       Acer             Aspire SW3-016     2     ssd       32         10.1   
83        Acer     A315-31CDC UN.GNTSI.001    2     ssd      500         15.6   
108       Acer  Aspire ES-15 NX.GKYSI.010     4     hdd      500         15.6   
100       Acer    A315-31-P4CRUN.GNTSI.002    4     hdd      500         15.6   
73        Acer             Aspire ES1-523     4     hdd     1024         15.6   
..         ...                         ...  ...     ...      ...          ...   
154  Microsoft      Surface Book CR9-00013    8     ssd      128         13.5   
31        Acer                 Predator 17   16     ssd      256         17.3   
96   Alienware          AW13R3-7000SLV-PUS    8     ssd      256         13.3   
90   Alienware                 15 Notebook   16     hdd     1024         15.6   
5        Apple      MacBook Pro (TouchBar)   16     ssd      512         15.0   

      price processor_brand processor_model  clock_speed graphic_card_brand  \
148   13872           intel      Atom Z8300         1.44              intel   
83    17990           intel         Celeron         1.10              intel   
108   17990             amd         A4-7210         1.80                amd   
100   18990           intel         pentium         1.10              intel   
73    19465             amd         A4-7210         1.80                amd   
..      ...             ...             ...          ...                ...   
154  178799           intel              i5         1.80              intel   
31   178912           intel              i7         2.60             nvidia   
96   190256           intel              i7         3.00             nvidia   
90   199000           intel              i7         2.60             nvidia   
5    226000           intel              i7         2.70              intel   

     graphic_card_size       os  weight  \
148                NaN  windows     1.2   
83                 NaN  windows     2.1   
108                NaN  windows     2.4   
100                NaN  windows     NaN   
73                 NaN    linux     2.4   
..                 ...      ...     ...   
154                NaN  windows     1.5   
31                 NaN  windows     4.2   
96                 6.0  windows     2.6   
90                 8.0  windows     3.5   
5                  2.0      mac     2.5   

                                              comments  
148                                                NaN  
83                                                 NaN  
108                                                NaN  
100                                                NaN  
73                                                 NaN  
..                                                 ...  
154                                                NaN  
31                                 Integrated Graphics  
96   13.3 inch FHD (1920 x 1080) IPS Anti-Glare 300...  
90      Maximum Display Resolution : 1920 x 1080 pixel  
5                                                  NaN  

[167 rows x 15 columns]

In [19]:
laptops_df

brand             model  ram hd_type  hd_size  screen_size   price  \
0     Dell  Inspiron 15-3567    4     hdd     1024         15.6   40000   
1    Apple       MacBook Air    8     ssd      128         13.3   55499   
2    Apple       MacBook Air    8     ssd      256         13.3   71500   
3    Apple       MacBook Pro    8     ssd      128         13.3   96890   
4    Apple       MacBook Pro    8     ssd      256         13.3  112666   
..     ...               ...  ...     ...      ...          ...     ...   
162   Asus            A555LF    8     hdd     1024         15.6   39961   
163   Asus     X555LA-XX172D    4     hdd      500         15.6   28489   
164   Asus            X554LD    2     hdd      500         15.6   29199   
165   Asus    X550LAV-XX771D    2     hdd      500         15.6   29990   
166   Asus     X540LA-XX538T    4     hdd     1024         15.6   30899   

    processor_brand processor_model  clock_speed graphic_card_brand  \
0             intel              i5          2.5              intel   
1             intel              i5          1.8              intel   
2             intel              i5          1.8              intel   
3             intel              i5          2.3              intel   
4             intel              i5          2.3              intel   
..              ...             ...          ...                ...   
162           intel      i3 4th gen          1.7             nvidia   
163           intel      i3 4th gen          1.9              intel   
164           intel      i3 4th gen          1.9              intel   
165           intel      i3 4th gen          1.7              intel   
166           intel      i3 5th gen          2.0              intel   

     graphic_card_size       os  weight comments  
0                  NaN    linux    2.50      NaN  
1                  2.0      mac    1.35      NaN  
2                  2.0      mac    1.35      NaN  
3                  2.0      mac    3.02      NaN  
4                  2.0      mac    3.02      NaN  
..                 ...      ...     ...      ...  
162                2.0  windows    2.30      NaN  
163                NaN    linux    2.30      NaN  
164                1.0    linux    2.30      NaN  
165                NaN    linux    2.50      NaN  
166                NaN  windows    2.30      NaN  

[167 rows x 15 columns]

In [21]:
laptops_df.sort_values(by="price", ascending=False, inplace=True)
laptops_df

brand                       model  ram hd_type  hd_size  screen_size  \
5        Apple      MacBook Pro (TouchBar)   16     ssd      512         15.0   
90   Alienware                 15 Notebook   16     hdd     1024         15.6   
96   Alienware          AW13R3-7000SLV-PUS    8     ssd      256         13.3   
31        Acer                 Predator 17   16     ssd      256         17.3   
154  Microsoft      Surface Book CR9-00013    8     ssd      128         13.5   
..         ...                         ...  ...     ...      ...          ...   
73        Acer             Aspire ES1-523     4     hdd     1024         15.6   
100       Acer    A315-31-P4CRUN.GNTSI.002    4     hdd      500         15.6   
108       Acer  Aspire ES-15 NX.GKYSI.010     4     hdd      500         15.6   
83        Acer     A315-31CDC UN.GNTSI.001    2     ssd      500         15.6   
148       Acer             Aspire SW3-016     2     ssd       32         10.1   

      price processor_brand processor_model  clock_speed graphic_card_brand  \
5    226000           intel              i7         2.70              intel   
90   199000           intel              i7         2.60             nvidia   
96   190256           intel              i7         3.00             nvidia   
31   178912           intel              i7         2.60             nvidia   
154  178799           intel              i5         1.80              intel   
..      ...             ...             ...          ...                ...   
73    19465             amd         A4-7210         1.80                amd   
100   18990           intel         pentium         1.10              intel   
108   17990             amd         A4-7210         1.80                amd   
83    17990           intel         Celeron         1.10              intel   
148   13872           intel      Atom Z8300         1.44              intel   

     graphic_card_size       os  weight  \
5                  2.0      mac     2.5   
90                 8.0  windows     3.5   
96                 6.0  windows     2.6   
31                 NaN  windows     4.2   
154                NaN  windows     1.5   
..                 ...      ...     ...   
73                 NaN    linux     2.4   
100                NaN  windows     NaN   
108                NaN  windows     2.4   
83                 NaN  windows     2.1   
148                NaN  windows     1.2   

                                              comments  
5                                                  NaN  
90      Maximum Display Resolution : 1920 x 1080 pixel  
96   13.3 inch FHD (1920 x 1080) IPS Anti-Glare 300...  
31                                 Integrated Graphics  
154                                                NaN  
..                                                 ...  
73                                                 NaN  
100                                                NaN  
108                                                NaN  
83                                                 NaN  
148                                                NaN  

[167 rows x 15 columns]

# 큰 Series 살펴보기

In [23]:
laptops_df['brand']

5          Apple
90     Alienware
96     Alienware
31          Acer
154    Microsoft
         ...    
73          Acer
100         Acer
108         Acer
83          Acer
148         Acer
Name: brand, Length: 167, dtype: object

* 몇개의 브랜드가 있는지 보고 싶어!
* `series.unique()`

In [25]:
laptops_df['brand'].unique()

array(['Apple', 'Alienware', 'Acer', 'Microsoft', 'HP', 'Dell', 'Lenovo',
       'Asus'], dtype=object)

* 각 브랜드마다 몇번 나오는지 궁금해!
* `series.value_counts()`

In [27]:
laptops_df['brand'].value_counts()

HP           55
Acer         35
Dell         31
Lenovo       18
Asus          9
Apple         7
Alienware     6
Microsoft     6
Name: brand, dtype: int64

* 특정 series의 기본 정보(요약본)을 알고 싶다면
* `series.decribe()`

In [28]:
laptops_df['brand'].describe()

count     167
unique      8
top        HP
freq       55
Name: brand, dtype: object